### DISCUSSION TITLE
esp32 behaves differently to esp32s2 with esp32.wake_on_ext0()

### DISCUSSION THREAD

**Discussion by kjm1102** [micropython/micropython#15252] [ESP32]
I'm finding
```
import os,time,esp32
from machine import Pin,deepsleep

print (os.uname())
time.sleep(5)
pb=Pin(0, Pin.IN); esp32.wake_on_ext0(pb,0)
deepsleep(10*1000)
```
works on the esp32 (the board deepsleeps then wakes after 10s, or earlier if I press the button) but not the esp32s2 (exits deepsleep immediately after entering it). Both boards using upython v1.23.0 although this is not unique to that version.

Since this behaviour is also influenced by the API, the esp32 behaves properly in some but not in others, the best way to validate is to save the file as main.py on the board then reset it without an api (ie the usb is providing power only no comms). 

Since the 2 boards I was testing with had no user led in common & I was keen to run exactly the same code on both, I put a usb ammeter in series. On the usb ammeter I could see the esp32 current drop when it went into deepsleep then rise if I pressed the button or waited 10s. With the esp32s2 the current remained pretty much constant since the s2 quit deepsleep immediately after entering.

Has anyone else experienced this problem or seen it in the bug tracker? Should we raise it as an issue?

For background please see https://github.com/orgs/micropython/discussions/14539 and https://github.com/orgs/micropython/discussions/15234

**Comment by scruss** [comment]
Have you tried [ThinkTransit](https://github.com/orgs/micropython/discussions/15234#discussioncomment-9751258)'s suggestion of enabling the built-in pull-up resistor?

**Reply by kjm1102** [reply]
I did Stew, but it still jumps out of deepsleep immediately. Something knocks gpio0 permanently lo when I deepsleep. I've got a hunch that the pullup is not maintained in deepsleep. I know hold is supposed to be for outputs only but I might try it with the input definition for gpio0. If that doesn't work I'll try adding an external pullup.
